src/models/mSeriesChallenge.ts

```diff
@@ -0,0 +1,268 @@
+   * @param options.sort default: 'name'
+   * @param options.ascending default: true
+   * @param options.limit default: 20
+   * @param options.offset default: 0
+   * @returns An array of MSeries Challenge Participants
+   */
+  async getParticipants (options: {
+    nameSearchQuery: string
+    sort?: MSeriesChallengeParticipantSorting
+    ascending?: boolean
+    limit?: number
+    offset?: number
+  }) {
+    const participants = await this.action('mSeriesChallengeParticipant:list', {
+      mSeriesChallengeId: this._mSeriesChallengeData.id,
+      userId: this.sessionHandler.userId,
+      ...options
+    }) as MSeriesChallengeParticipantListResponse
+    return new MSeriesChallengeParticipants(participants.mSeriesChallengeParticipants, participants.mSeriesChallengeParticipantsMeta, this.sessionHandler)
+  }
+
+  /**
+   * Method to get leaderboard of challenge.
+   * Session User must either own or have joined challenge to get leaderboard.
+   *
+   * @param options.targetUserId userId to "center" leaderboard on
+   * @param options.ascending default: true
+   * @param options.limit default: 20
+   * @param options.offset default: 0
+   * @returns An array of MSeries Challenge Participants sorted by rank
+   */
+  async getLeaderboard (options: {
+    targetUserId?: number
+    ascending?: boolean
+    limit?: number
+    offset?: number
+  } = {}) {
+    const leaderboard = await this.action('mSeriesChallenge:leaderboard', {
+      id: this._mSeriesChallengeData.id,
+      userId: this.sessionHandler.userId,
+      ...options
+    }) as MSeriesChallengeLeaderboardResponse
+
+    return new MSeriesChallengeLeaderboardParticipants(leaderboard.mSeriesChallengeParticipants, leaderboard.mSeriesChallengeParticipantsMeta, this.sessionHandler)
+  }
+
+  /**
+   * Method to update challenge meta data.
+   * Session User must own challenge to update it.
+   *
+   * @param options.name
+   * @param options.userLimit
+   */
+  async update (options: {
+    name?: string
+    // isPublic?: boolean
+    userLimit?: number
+  }) {
+    await this.action('mSeriesChallenge:update', {
+      id: this._mSeriesChallengeData.id,
+      ...options
+    }) as MSeriesChallengeResponse
+
+    await this.reload()
+  }
+
+  get id () {
+    return this._mSeriesChallengeData.id
+  }
+
+  get joinCode () {
+    return this._mSeriesChallengeData.joinCode
+  }
+
+  get userId () {
+    return this._mSeriesChallengeData.userId
+  }
+
+  get name () {
+    return this._mSeriesChallengeData.name
+  }
+
+  get isPublic () {
+    return this._mSeriesChallengeData.isPublic
+  }
+
+  get userLimit () {
+    return this._mSeriesChallengeData.userLimit
+  }
+
+  get challengeType () {
+    return this._mSeriesChallengeData.challengeType
+  }
+
+  get startAt () {
+    return new Date(this._mSeriesChallengeData.startAt)
+  }
+
+  get endAt () {
+    return this._mSeriesChallengeData.endAt !== null ? new Date(this._mSeriesChallengeData.endAt) : null
```

Comment:
Should use nullish coalescing

Reply:
Not exactly sure what you're proposing here. 

```
return new Date(this._mSeriesChallengeData.endAt) ?? null 
```
would not work correctly. `new Date(null)` returns a valid date, which is not desirable here.

Reply:
My bad, you're correct.